app/graphql/mutations/remove_reaction.rb

```diff
@@ -11,6 +11,18 @@ def resolve(_params)
       { success: true }
     end
 
+    class ValidateReactionExists < GraphQL::Schema::Validator
+      def validate(_object, _context, value)
+        if !Reaction.exists?(id: value[:reaction_id])
+          return(
+            I18n.t("mutations.remove_reaction.reaction_not_found")
+          )
+        end
+      end
+    end
+
+    validates ValidateReactionExists => {}
```

Comment:
I don't think this validation is required.

The linked issue which you wrote says this under _Expected Behavior_:

> The query_authorized? method should handle the case where the reaction is nil and return false or raise a more descriptive error indicating that the reaction was not found..

I think it'd be easier for the mutation to simply pretend to delete a `Reaction` when one doesn't exist. I.e., always say `success: true`, regardless of whether the supplied reaction exists or not.

This is overall simpler; consider the example you gave in the linked issue of two tabs being open and _remove reaction_ being triggered from both. If the mutation had pretended to work, then the UI in both tabs would have been fine with it. It's just that only one would have actually deleted the `reactions` table entry.

The catch-all in `CoursesCurriculum__Reactions`...

```res
Js.Promise2.catch(_error => {
  Js.Promise.resolve()
})
```

...probably isn't required either. Would we really want to catch any error and squash it? That doesn't seem right.

Reply:
> The query_authorized? method should handle the case where the reaction is nil and return false or raise a more descriptive error indicating that the reaction was not found..
> I think it'd be easier for the mutation to simply pretend to delete a Reaction when one doesn't exist. I.e., always say success: true, regardless of whether the supplied reaction exists or not.

Though about it, but decided to do with validator 😁.

---

> This is overall simpler; consider the example you gave in the linked issue of two tabs being open and remove reaction being triggered from both. If the mutation had pretended to work, then the UI in both tabs would have been fine with it. It's just that only one would have actually deleted the reactions table entry.

Yes, you are right, it is simpler